tests/unit/withOnyxTest.js

```diff
@@ -522,4 +523,98 @@ describe('withOnyxTest', () => {
                 expect(onRender.mock.calls[3][0].simple).toBe('long_string');
             });
     });
+
+    it('should render immediately when a onyx component is mounted a 2nd time', () => {
+        const TestComponentWithOnyx = withOnyx({
+            text: {
+                key: ONYX_KEYS.TEST_KEY,
+            },
+        })(ViewWithText);
+        const onRender = jest.fn();
+        let renderResult;
+
+        // Set the value in storage, but not in cache.
+        return StorageMock.setItem(ONYX_KEYS.TEST_KEY, 'test1')
+            .then(() => {
+                renderResult = render(<TestComponentWithOnyx onRender={onRender} />);
+
+                // The component should not render initially since we have no data in cache.
+                expect(onRender).not.toHaveBeenCalled();
+                return waitForPromisesToResolve();
+            })
+            .then(waitForPromisesToResolve)
+            .then(() => {
+                let textComponent = renderResult.getByText('test1');
+                expect(textComponent).not.toBeNull();
+                expect(onRender).toHaveBeenCalledTimes(1);
+
+                // Unmount the component and mount it again. It should now render immediately.
```

Comment:
Yes, if the assertion passes right away, before `waitForPromisesToResolve` this means it rendered synchronously since all data is cached. I can clarify.

Reply:
+1